### PR TITLE
rose edit: extended multiple selection to groups (e.g. stash groups)

### LINKED ISSUE #1410: rose edit: um widget: ignore/enable by group
It would be good to be able to apply actions on a number of grouped stash
requests at once. This is likely to be just ignore/enable.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1410.
Follow-on from #1859 
Solves right click issue (https://github.com/metomi/rose/pull/1859#issuecomment-198287340)

Stash entries can now be ignored/deleted for a tree of entries (i.e. in group view). If a parent section is ignored/deleted then this action will apply (recursively) to all child sections.

@matthewrmshin @arjclark Please Review.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@stevewardle has agreed to sanity check and test this for us.

**Comment by stevewardle** [comment]
Code looks fine to me - I also gave it a good testing with some apps and it appears to do what it says, and intuitively what I would expect it to

**Comment by arjclark** [comment]
As discussed, two things to investigate:
 * the multiple ignores/deletes/whatevers should go on the undo stack as a single operation
 * oddity with ignoring groups - crash bug fixed here (yay!) but inconsistent resulting display

**Comment by oliver-sanders** [comment]
b1cb03313b57feb6789cfda5ef345fb2ea28092e - Addresses the two bullets from the previous comment.

Behaviour change: When using `Ctrl-I` to toggle ignore selected sections the previous behaviour was to toggle each setting individually. The new behaviour is to ignore all sections unless they are all already ignored in which case they should all be enabled.

File: lib/python/rose/config_editor/panelwidget/summary_data.py [hunk 1]
```diff
@@ -536,6 +626,47 @@ def copy_section(self, section):
...
+        sections_ = self._get_selected_sections()
+        ignored = [rose.variable.IGNORED_BY_USER in
+                   self.sections[section_].ignored_reason for
+                   section_ in sections_]
+        # If ignore mode is not specified decide whether to ignore or enable.
+        if ignore is None:
+            if all(ignored):
+                ignore = False
+            else:
+                ignore = True
```
Comment (matthewrmshin) [inline lib/python/rose/config_editor/panelwidget/summary_data.py]: `ignore = not all(ignored)`?

**Comment by arjclark** [comment]
Looks good to me. Quirks tested as now working.

**Pull request merged**